### ISSUE TITLE
Text is squiggly when writing in any input field

### ISSUE THREAD

**Issue by DeluxeOwl** [fyne-io/fyne#2206]
### Describe the bug:
This happens when I run the fyne_demo:
When I type in an input field that accepts text, the text is moving in a squiggly fashion. (Widgets -> Entry)

### To Reproduce:
Steps to reproduce the behaviour:
1. Download the fyne demo 
2. Go to any tab that has input (I chose Widgets -> Entry]
3. Type some text in a field
4. Observe the squiggly text

### Video:
[I've uploaded a short (11 sec) unlisted video here that shows the issue](https://youtu.be/So6EAMRMW0k)

### Device (please complete the following information):
 - **OS:** Manjaro Linux KDE using Xorg
 - **Version:** 5.11.16-2-MANJARO
 - **Go version:** go version go1.16.3 linux/amd64 (it happened in previous versions as well)
 - **Fyne version:** latest (v2)

**Comment by Jacalz** [comment]
Hmm, that’s an interesting one. Nothing that I have seen before.

Btw, I changed the Fyne version from ”latest” to 2.0.2. Using latest means that it is going to get old as soon as a new release is out.

**Comment by andydotxyz** [comment]
Yes in some systems I have seen this as well. It was mostly fixed by #1968, but it seems there is an underlying issue that was worked around instead of finding the cause.
